js/background.js

```diff
@@ -144,6 +146,10 @@
     }
 
     var language = detectLanguage(contentType, details.url);
+    if ((['html', 'xml'].indexOf(language) != -1) && (contentType === null)) {
```

Comment:
You're right. To be honest, instead of refactoring everything I was just "working around" what you had. Really just treading lightly on someone else's code. Maybe I shouldn't do that sometimes :) I'm glad you decided to go deeper with the fix. That's something  I was a little hesitant to suggest.